docs: Add BGD related changes in docs

kannann1 opened the pull request.

kannann1 committed:
doc: add namespace render filtering

popoveugene committed:
docs: add clarification for bg_domain

popoveugene committed:
docs: update bg domain object

popoveugene committed:
docs: minor

kannann1 committed:
chore: add template macros for ns filtering

popoveugene committed:
docs: update bg controller cred

popoveugene committed:
docs: add `bg_domain.controllerNamespace.credentialsValue` to effective set

popoveugene committed:
docs: add bg domain cred details

popoveugene committed:
Merge branch 'main' into doc/ns-filter

popoveugene committed:
docs: cred update

kannann1 committed:
chore: address review comments

kannann1 committed:
Merge branch 'main' into doc/ns-filter

kannann1 committed:
chore: add BG related macros for Blue-Green deployments

Added BG related macros for Blue-Green Domain deployments, including descriptions, types, and basic usage examples for each macro.

kannann1 committed:
chore: add BG related macros for Blue-Green deployments

kannann1 committed:
chore: add controller, origin, and peer to macros list

popoveugene committed:
docs: update ns filter macro

popoveugene committed:
docs: remove Namespace Filter Macros from template macros

popoveugene committed:
docs: updated bg related macro

popoveugene reviewed (changes requested):
1. In all docs where the BG domain object is mentioned, replace credentialsId with credentials. Also, create a GitHub ticket for the effective set calculator so that it correctly generates bg_domain in the effective set.

2. I don't understand the description of the following macros. It looks like the calculator computes them differently. Please clarify with Chethana.

    - BG_CONTROLLER_URL
    - CMDB_CALLBACK_URL
    - CMDB_CALLBACK_TOKEN

3. I changed the NS filter macro from ${} to @.

4. I removed the NS filter from template macros. I was wrong when I suggested adding it.
    The descriptions in namespace-render-filtering.md and instance-pipeline-parameters.md are sufficient.

5. I made minor changes: https://github.com/Netcracker/qubership-envgene/commit/7c95ce643088ec9350248bf8695781c9dcdc7460

kannann1 committed:
fix: address review comment

popoveugene committed:
docs:  GITHUB_PIPELINE_API_INPUT-> GH_ADDITIONAL_PARAMS

popoveugene committed:
docs: test

kannann1 committed:
chore: replace credentialsId with credentials for bg_domain object

kannann1 committed:
chore: remove CMDB_CALLBACK_URL and CMDB_CALLBACK_TOKEN

Removed CMDB_CALLBACK_URL and CMDB_CALLBACK_TOKEN sections from the documentation, [corresponding issue](https://github.com/Netcracker/qubership-envgene/issues/745) has been created to remove these from the them from the code.

popoveugene committed:
Merge branch 'main' into doc/ns-filter

popoveugene committed:
docs: update bg related macros

popoveugene committed:
docs: remove credentials

popoveugene committed:
docs: add bg_manage

popoveugene committed:
docs: wip

popoveugene committed:
docs: fix link

popoveugene committed:
docs: add Namespace Folder Name Generation Rules

popoveugene committed:
docs: rename deploypostfix folder to ns folder

popoveugene committed:
docs: add deployPostfix Matching Logic + change topology's `environments` var generation

popoveugene committed:
docs: add separate template artifacts for origin and peer namespaces

popoveugene committed:
docs add use cases

popoveugene committed:
docs: add UC

popoveugene committed:
Merge branch 'main' into doc/ns-filter

popoveugene committed:
docs: fix liners
